snap/errors.go

```diff
@@ -43,11 +43,6 @@ func (e NotInstalledError) Error() string {
 	return fmt.Sprintf("revision %s of snap %q is not installed", e.Rev, e.Snap)
 }
 
-func (e *NotInstalledError) Is(err error) bool {
-	_, ok := err.(*NotInstalledError)
-	return ok
-}
-
```

Comment:
is this intentional or a problem with the merge?

Reply:
this can be used via errors.Is though, are we sure is not used indirectly like that?

Reply:
Actually it is in devicestate, thanks for noticing. I've changed now the test so `NotInstalledError.Is` is now used indirectly via errors to make this clearer. This made me thing a little bit though, as in the Is implementation we are comparing only the type and we can have surprises as, for instance, this test would fail:
```
func (s *errorsSuite) TestNotInstalledErrorIs2(c *C) {
	err := snap.NotInstalledError{"foo", snap.R(10)}
	c.Check(errors.Is(err, snap.NotInstalledError{}), Equals, true)
	c.Check(errors.Is(errors.New("some error"), snap.NotInstalledError{}), Equals, false)
}
```
which is not consistent with what we have (could be fixed though with a more ellaborate implementation of `NotInstalledError.Is`). Maybe we should start to use `errors.As()` more in the future if what we really want to compare is error types, not the full error content?

Reply:
yes, this "abuse" of Is seems a pattern that we have grown independently as a bit of middle ground between Is and As. I didn't spot immediately when we started it. We have a few *Error types that defines such a Is. I agree it could be better to move away from it but it's probably something to discuss with the team

Reply:
It does seem that `As` is more appropriate to use for these cases where we don't care about the contents of the custom error type.